pkg/prometheus/server/operator.go

```diff
@@ -940,8 +942,17 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 			return
 		}
 
+		shouldDelete, err := c.shouldDelete(p)
```

Comment:
(nit) it might be easier to read if the function is `shouldRetain` (e.g. no need to negate the returned value to exit early).